src/runtime_src/core/tools/xbflash2/OO_Program_Qspips.cpp

```diff
@@ -0,0 +1,295 @@
+/**
+ * Copyright (C) 2022 Licensed under the Apache License, Version
+ * 2.0 (the "License"). You may not use this file except in
+ * compliance with the License. A copy of the License is located
+ * at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+// ------ I N C L U D E   F I L E S -------------------------------------------
+// Local - Include Files
+#include "OO_Program_Qspips.h"
+#include "tools/common/XBUtilitiesCore.h"
+
+#include "core/pcie/tools/xbflash.qspi/firmware_image.h"
+#include "core/pcie/tools/xbflash.qspi/pcidev.h"
+#include "core/pcie/tools/xbflash.qspi/xqspips.h"
+
+namespace XBU = XBUtilities;
+
+// 3rd Party Library - Include Files
+#include <boost/program_options.hpp>
+#include <boost/format.hpp>
+#include <boost/algorithm/string.hpp>
+namespace po = boost::program_options;
+
+// System - Include Files
+#include <iostream>
+#include <fstream>
+
+namespace {
+
+int 
+qspips_flash(po::variables_map& vm) {
+    std::string bdf;
+    std::string flash_type;
+    std::string bin_file;
+    std::vector<std::string> bin_files;
+    std::string soffset;
+    size_t offset = 0;
+    bool force = false;
+    std::string sBar, sBarOffset;
+    size_t baroff = INVALID_OFFSET;
+    int bar = 0;   
+
+    XBU::sudo_or_throw_err();
+     
+    //mandatory command line args
+    try {
+        bdf = vm["device"].as<std::string>();
+        bin_files = vm["image"].as<std::vector<std::string>>();
+    }
+    catch (...) {
+        throw std::errc::invalid_argument;
+    }
+
+    //optional command line args
+    try {
+        if (vm.count("flash-part"))
+            flash_type = vm["flash-part"].as<std::string>();
+        if (vm.count("offset")) {
+            soffset = vm["offset"].as<std::string>();
+            if (!soffset.empty()) {
+                std::stringstream sstream(soffset);
+                sstream >> offset;
+            }
+        }
+        if (vm.count("force"))
+            force = true;
+    }
+    catch (...) {
+    }
+
+    //optionals command line args
+    try {
+        if (vm.count("bar")) {
+            sBar = vm["bar"].as<std::string>();
+            if (!sBar.empty())
+                bar = std::stoi(sBar);
+        }
+        if (vm.count("bar-offset")) {
+            sBarOffset = vm["bar-offset"].as<std::string>();
+            if (!sBarOffset.empty()) {
+                std::stringstream sstream(sBarOffset);
+                sstream >> baroff;
+            }
+        }
+    }
+    catch (...) {
+    }
+
+    if (bdf.empty()) {
+        std::cout << "Error: Please provide mgmt BDF\n";
```

Comment:
If this is an error, please use std::err.  Also consider throwing the exception instead of printing it.